ci/jupyter.js

```diff
@@ -6,12 +6,41 @@ const { sleep } = require("./sleep");
 global.XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest;
 const { ajax } = require("rxjs/ajax");
 
+let url_path_join = function(...pieces) {
+  // """Join components of url into a relative url
+  // Use to prevent double slash when joining subpath. This will leave the
+  // initial and final / in place
+  // """
+  initial = pieces[0].startsWith("/");
+  final = pieces[pieces.length - 1].endsWith("/");
+  const hi = [];
+  for (s of pieces) {
```

Comment:
Did you know you can use `Array.map()` to achieve the same thing here? Although, I do believe a `for` loop is more performant. Here is a [blog post](https://medium.com/@ExplosionPills/map-vs-for-loop-2b4ce659fb03) that talks about using `for` vs `map`. Again, not required. But generally `map` is used in replace of a `for` loop.

Reply:
Sounds good!